rocketpy/mathutils/function.py

```diff
@@ -3261,12 +3353,19 @@ def __validate_extrapolation(self, extrapolation):
 
         ## multiple dimensions
         elif self.__dom_dim__ > 1:
-            if extrapolation not in [None, "natural"]:
+            if extrapolation is None:
+                extrapolation = "natural"
+            if extrapolation == "natural" and self.__interpolation__ == "linear":
+                warnings.warn(
+                    "Extrapolation 'natural' is not supported for multidimensional "
+                    "linear interpolation. 'rbf' will be used to extrapolate."
+                )
+            elif extrapolation not in ["constant", "natural", "zero"]:
                 warnings.warn(
                     "Extrapolation method set to 'natural'. Other methods "
                     "are not supported yet."
                 )
-            extrapolation = "natural"
+                extrapolation = "natural"
         return extrapolation
```

Comment:
Missing a check for `"rbf"` in this metod. Rigth now setting extrapolation to rbf with interpolation linear raises a warning

Reply:
Yes, indeed, this happens because "rbf" is not a valid extrapolation input: the same way one cannot input "linear" or "spline" for 1-D Functions, a warning will also occur.

As it is for 1-D Functions, the only extrapolation inputs are "constant", "natural" and "zero".

Reply:
The warning message says: `... 'rbf' will be used to extrapolate."`. If rbf is not an extrapolation method, this warning is wrong.

This warning is also raised if interpolation="linear" and extrapolation="natural"

Reply:
The warning should _only_ be raised if interpolation is "linear" and extrapolation "natural". The warning is there to show that a 'rbf' algorithm is used to extrapolate in this case. The reasoning behind it I have explained in the PR description and is detailed in scipy issue 6396.

I can change the warning to be more precise and better phrased, like:

```
"Extrapolation 'natural' for multidimensional linear "
"interpolation uses a 'rbf' algorithm for smoother results."
```

Reply:
I don't think there should be warning at all in this case. Just mention this behaviour in the docstrings.

With the way it is, there will be warnings whenever we define a ND function inside the code. So there will be a lot of warnings for a desired behaviour

Reply:
Makes total sense, I believe the warning was a bit overzealous in this case. I introduced this behavior in commit [bae7cae](https://github.com/RocketPy-Team/RocketPy/pull/691/commits/bae7caef22ac94a103c1ed345a5654e4072b24d5).